nixos/tests/podman.nix

```diff
@@ -53,22 +53,11 @@ import ./make-test-python.nix (
           podman.succeed("podman stop sleeping")
           podman.succeed("podman rm sleeping")
 
-      with subtest("Run container rootless with runc"):
+      with subtest("Run container rootless"):
           podman.succeed(su_cmd("tar cv --files-from /dev/null | podman import - scratchimg"))
           podman.succeed(
               su_cmd(
-                  "podman run --runtime=runc -d --name=sleeping -v /nix/store:/nix/store -v /run/current-system/sw/bin:/bin scratchimg /bin/sleep 10"
-              )
-          )
-          podman.succeed(su_cmd("podman ps | grep sleeping"))
-          podman.succeed(su_cmd("podman stop sleeping"))
-          podman.succeed(su_cmd("podman rm sleeping"))
-
-      with subtest("Run container rootless with crun"):
-          podman.succeed(su_cmd("tar cv --files-from /dev/null | podman import - scratchimg"))
-          podman.succeed(
-              su_cmd(
-                  "podman run --runtime=crun -d --name=sleeping -v /nix/store:/nix/store -v /run/current-system/sw/bin:/bin scratchimg /bin/sleep 10"
+                  "podman run -d --name=sleeping -v /nix/store:/nix/store -v /run/current-system/sw/bin:/bin scratchimg /bin/sleep 10"
```

Comment:
Can we keep the explicit `--runtime=crun`/ `Run container rootless with crun`  please?

Reply:
Can you elaborate on why? `podman` seems to default to it with cgroupsv2, and if we view this test from a user perspective, they just want to `podman run` as non-root, without explicitly configuring a specific runtime.

also cc @adisbladis

Reply:
I just want to say I agree with @flokli on this one. It's better that we test the users default config.
Also I seem to recall seeing some intent to switch back to `runc` by default at some point once it's gained the required cgroupsv2 support (don't quote me on this one).

Reply:
Can we do this instead?

```diff
diff --git a/nixos/tests/podman.nix b/nixos/tests/podman.nix
index cd8c2b4308c..74581f5c847 100644
--- a/nixos/tests/podman.nix
+++ b/nixos/tests/podman.nix
@@ -55,13 +55,13 @@ import ./make-test-python.nix (
 
       with subtest("Run container rootless with runc"):
           podman.succeed(su_cmd("tar cv --files-from /dev/null | podman import - scratchimg"))
-          podman.succeed(
+          podman.fail(
               su_cmd(
                   "podman run --runtime=runc -d --name=sleeping -v /nix/store:/nix/store -v /run/current-system/sw/bin:/bin scratchimg /bin/sleep 10"
               )
           )
-          podman.succeed(su_cmd("podman ps | grep sleeping"))
-          podman.succeed(su_cmd("podman stop sleeping"))
+          podman.fail(su_cmd("podman ps | grep sleeping"))
+          podman.fail(su_cmd("podman stop sleeping"))
           podman.succeed(su_cmd("podman rm sleeping"))
 
       with subtest("Run container rootless with crun"):

```

Reply:
I don't think our test should suddenly fail if podmans runc backend starts working in rootless scenarios again.

I'm fine adding a test for the default backend in as-root scenarios, but made the rootless runc subtest a comment, which we can re-add once it gets implemented.

Reply:
> I don't think our test should suddenly fail if podmans runc backend starts working in rootless scenarios again.
> 
> I'm fine adding a test for the default backend in as-root scenarios, but made the rootless runc subtest a comment, which we can re-add once it gets implemented.

All of these packages have `passthru.tests` so the failure will be noticed when it does start working.

Can you move the default tests after the explict runtime tests please? If it does fail because of a runtime problem having it fail on the explict test would be clearer.

Reply:
I can move the test order, sure, but I still don't think the test should suddenly fail if *more* things work. `podman` is still functional if a backend that was known to be broken starts working again. We shouldn't then fail the test.

Reply:
I'd expect the maintainers of this package to read the release notes while bumping, and updating/extending the test accordingly once new features are added.